### PR TITLE
AnswerModal implemented

### PR DESCRIPTION
**Pull request by ME2SWE**
AnswerModal completed with thumbnail and upload image functionality. Had a funky behavior w/ Answermodal while no such behavior w/ QuestionModal. I had to add a couple of modifications to make it work.

### CODE REVIEW

**Review by saikitjk — APPROVED** [review]
good work

File: client/src/components/QA/AnswerModal.jsx [hunk 1]
```diff
@@ -27,13 +27,68 @@ class AnswerModal extends React.Component {
...
     };
     this.handleSubmitError = this.handleSubmitError.bind(this);
+    this.handleImageUpload = this.handleImageUpload.bind(this);
+    this.handleInput = this.handleInput.bind(this);
   };
 
+  handleInput(e) {
+    var type = e.target.id;
+    var value = e.target.value;
+    if (type === 'Email') {
```
Comment (LesliePajuelo) [inline client/src/components/QA/AnswerModal.jsx:45]: If you coerce your type to be lower-case and use an [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) to be sure it's a valid "type" and then set state directly
```
validTypes.includes(type) ? 
this.setState({[type]: value}) : this.setState({error: {type: "validation", message: `incorrect type, ${type}`}})`

File: client/src/components/QA/AnswerModal.jsx [hunk 2]
```diff
@@ -67,26 +122,39 @@ class AnswerModal extends React.Component {
...
 
   render() {
     return(
       <div className='answer-modal-container'>
         <Modal
           open={this.props.isAModalOpen}
           onClose={() => {this.props.handleAModalClose()}}
-          aria-labelledby='modal-modal-title'
-          aria-describedby='modal-modal-description'
+          aria-labelledby='modal-answer-modal-title'
```
Comment (LesliePajuelo) [inline client/src/components/QA/AnswerModal.jsx:149]: Good use of lablled by!

File: client/src/components/QA/AnswerModal.jsx [hunk 3]
```diff
@@ -95,49 +163,56 @@ class AnswerModal extends React.Component {
...
+                id='Email'
                 label='Email'
                 placeholder="example@atelier.com"
                 fullWidth
                 inputProps={{maxLength: 60}}
                 value={this.state.email}
                 error={!!this.state.emailError}
                 helperText={this.state.emailError}
-                onChange={e => this.setState({email: e.target.value})}
+                onChange={this.handleInput}
```
Comment (LesliePajuelo) [inline client/src/components/QA/AnswerModal.jsx:181]: I suggest you give this function a more meaningful name, we know that it's an on change, what does it mean to handle the input? Validate input?

File: client/src/components/QA/AnswerModal.jsx [hunk 4]
```diff
@@ -95,49 +163,56 @@ class AnswerModal extends React.Component {
...
                 helperText={this.state.nicknameError}
-                onChange={e => this.setState({nickname: e.target.value})}
+                onChange={this.handleInput}
                 required>
               </TextField>
               <p>For privacy reasons, do not use your full name or email address</p>
             </Stack>
             <br></br>
+            <Stack direction='row' spacing={1}>
+              {this.state.images.map((image, index) => <img id='thumbnail' src={image}></img>)}
```
Comment (LesliePajuelo) [inline client/src/components/QA/AnswerModal.jsx:203]: An `id` should only appear once per page! 

This should be changed to a class name. 

**Required change. Different browsers will handle this differently and unpredictably!**

**Review by LesliePajuelo — COMMENTED** [review]
Good work overall. It would be worth the time to have less repetitive code. 

I'm requiring you change the thumbnail id to a class name.  Send an email once completed with the PR with this change. @saikitjk

File: client/dist/styles.css [hunk 5]
```diff
@@ -430,5 +430,8 @@ main {
   cursor: pointer;
 }
 
-
+#thumbnail {
```
[the remark this answers is not in the document]
Comment (ME2SWE) [reply client/dist/styles.css:433]: Removed and using Jason's CSS classname instead.

File: client/src/components/QA/AnswerModal.jsx [hunk 6]
```diff
@@ -27,13 +27,68 @@ class AnswerModal extends React.Component {
       email:'',
       answerError: '',
       nicknameError: '',
-      emailError: ''
+      emailError: '',
```
[the remark this answers is not in the document]
Comment (ME2SWE) [reply client/src/components/QA/AnswerModal.jsx:30]: Tried but got stuck due to shallow-merging of setState. Tried implementing a few solutions but had no luck. Sticking w/the current state for now.

File: client/src/components/QA/AnswerModal.jsx (hunk 3 above)
Comment (ME2SWE) [reply client/src/components/QA/AnswerModal.jsx:181]: Replaced w/ a more meaningful name

File: client/src/components/QA/AnswerModal.jsx (hunk 4 above)
Comment (ME2SWE) [reply client/src/components/QA/AnswerModal.jsx:203]: Not using 'id' anymore.

**Pull request merged**